### ISSUE TITLE
Name on interfaces and blocks has not uniform height

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1635]
**Reported by massimo&#x2e;ceraolo on 21 Dec 2014 17:12 UTC**
The height of the %name field on MSL Modelica.Blocks.Interfaces.RealInput, Blocks.interfaces.BooleanInput and other input interfaces is 25, while the height of Blocks.Interfaces.Blocks.Icons.Block is 40.

Finally the height of other %name fields in MSL is again another value: Electrical.Interfaces.Pin %name text height is 60.

This creates a high inhomogeinity in diagrams, especially when tools do not override the tiny height of Input interface text height with larger values.

What I propose is to have a uniform height of all %name fields in MSL components, e.g. equal to the value in Blocks.Interfaces.Blocks.Icons.Block, i.e. 40

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1635

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 6 Jan 2015 16:04 UTC**
The "real" size of text depends on three numbers:
- The extent of the text (as mentioned in the ticket).
- The size of the coordinate system in which the text is drawn.
- The "scale" factor (when dragging a component in a diagram layer, the size of the component is "size of coordinate system" times "scale"; the default of scale = 0.1).

For connectors, another complication occurs:

 When dragging a connector in a diagram layer, it is displayed both in the icon and in the diagram layer where it is dragged. Unfortunately, Modelica has only ONE scale factor, both for icon and for diagram layer. If a connector would have exactly the same definition in its icon and diagram layer (coordinate system = {{-100,100},{-100,100}} and scale = 0.1), then it would be fine in the dragged diagram layer, but much too small in the dragged icon layer (dragging the icon of the new component would result in a much too small connector in this icon).

Due to this complication, the connectors in the Modelica Standard Library are designed in the following way:
- A default coordinate system is used both in icon and diagram layer, so {{-100,100},{-100,100}}.
- A non-default scale of 0.2 is used (instead of the default of 0.1).
- The drawing of the connector in the diagram layer is half the size of the drawing of the connector in its icon layer.

The effect is that when dragging this icon into the diagram layer of a new component, then the
- connector in the icon layer of this new component has size "0.2\*{{-100,100},{-100,100}})", that is it has the double size of a usual component.
- connector drawing in the diagram layer of this new component has practically a size of "0.1\*{{-100,100},{-100,100}})", whereas the text size is doubled as before (so a text defined with height "25", has actually a size of 0.2\*"25" = "5", instead of the default 0.1\*"25" = "2.5".

Experimenting with the text size defined in a connector was a compromise of the "visual" feeling. E.g. as you can see from the description above the RealInput "%name" text is in fact larger as the "%name" from a model component (it is equivalent to a 2\*25 = 50 height, instead of the 40 height in Icons.Block).

To summarize, designing the icon and diagram layer of a connector (including the %name text) is a really delicate topic and I do not want to change something, if there is not a concrete much better proposal for all the connectors in MSL.

**Comment by modelica-trac-importer** [comment]
**Modified by beutlich on 6 Jan 2015 21:31 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by massimo&#46;ceraolo on 9 Jan 2015 21:46 UTC**
Yes, I disregarded much of the complexity involved.
However what I wanted to stress is real and somewhat important.
Consider the following sequence:
1) launch a Modelica tool and open an empty model 
2) drag in a Modelica.Blocks.Interfaces.RealInput
3) drag in  a Modelica.Blocks.Continuous.Integrator.
4) Do not resize anything.

The result is very different in Dymola and OpenModelica.
In OpenModelica the interface will have the same size of the integrator and text on the interface is very tiny.
Opposite to this in Dymola the interface will have a larger size of the integrator and text on the interface is the same size of the integrator's.

Is there a mistake in one of the tools or there is a flaw in the MSL code of these two simple models?
